qcodes/utils/delaykeyboardinterrupt.py

```diff
@@ -47,8 +47,10 @@ def forceful_handler(sig, frame):
 
     def __exit__(self, exception_type: Optional[Type[BaseException]],
                  value: Optional[BaseException],
-                 traceback: Optional[TracebackType]):
+                 traceback: Optional[TracebackType]) -> None:
         if self.old_handler is not None:
             signal.signal(signal.SIGINT, self.old_handler)
         if self.signal_received is not None:
-            self.old_handler(*self.signal_received)
+            if (self.old_handler is not None and
+                    not isinstance(self.old_handler, int)):
```

Comment:
so if the old handler is `signal.SIG_DFL` then i understand that it's not possible to call it in this way `self.old_handler(*self.signal_received)` but perhaps we still should ask the `signal`'s machinery to somehow "simply perform the default function for the signal" (https://docs.python.org/3/library/signal.html#signal.SIG_DFL) ?

i have negligible clue of what i am talking about, btw :)

Reply:
I thoght about this. I would say that we should not do that. As the code currently is the user could have set another handler before entering the context so that calling the default handler would be wrong. I don't really understand when and how `signal.signal` would return None or an `int` rather than the old handler?

Reply:
then let's keep it as is :) as i said, i know almost nothing and also don't know "when and how signal.signal would return None or an int rather than the old handler" :)